src/redis-cli.c

```diff
@@ -2658,8 +2658,10 @@ static int parseOptions(int argc, char **argv) {
                 fprintf(stderr, "Invalid connection timeout for -t.\n");
                 exit(1);
             }
-            config.connect_timeout.tv_sec = (long long)seconds;
-            config.connect_timeout.tv_usec = ((long long)(seconds * 1000000)) % 1000000;
+            if (config.connect_timeout == NULL)
```

Comment:
how would someone explicitly set it to infinite?
maybe instead of NULL, we should have both struct members set to 0?

Reply:
not passing the struct means that timeout is not used. At this time, the system default is used (in this context is infinite).

0 seems to be a legal value (timeout immediately).

i am ok for `0 0` as no timeout

Reply:
> 0 seems to be a legal value (timeout immediately).

Really? :open_mouth: So it always times out then? It can never succeed immediately, can it?

> i am ok for `0 0` as no timeout

:+1:

Reply:
unlike `select` or `read` that can return data without blocking, i doubt it's a valid timeout for connect. let's use it in order to pass NULL to the API.